doc/source/whatsnew/v2.0.0.rst

```diff
@@ -165,6 +165,7 @@ Deprecations
 
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+- Removed deprecated :attr:`Timestamp.freq` (:issue:`14146`)
```

Comment:
Might be good to add `freqstr` attr was removed too, and that `freq` was removed from the `Timestamp` constructor and `Timestamp.fromordinal`